quantum/nodes/quantum-register/quantum-register.js

```diff
@@ -112,8 +117,10 @@ module.exports = function(RED) {
       // then send one qubit object per node output
       await shell.execute(script, (err) => {
         if (err) node.error(err);
-        else send(output);
       });
+      // wait for quantum circuit to be initialised
+      await circuitReady();
+      send(output);
```

Comment:
Probably want to change this to something like:
```node
// wait for quantum circuit to be initialised
await circuitReady();

// Run the script in the python shell, and if no error occurs
// then send one qubit object per node output
await shell.execute(script, (err) => {
  if (err) node.error(err);
  else send(output);
});
```
Ideally `send(output)` should be in the else of the `shell.execute` callback, since we don't want to continue the flow if an error has occurred (since doing so will inevitably result in other errors).

Reply:
This change will break the current node because the script for initialising quantum register should be executed before `await circuitReady();` is called, because inside the quantum register node, once it receives the message, it will update the `quantumCircuitConfig` object which stored in the flow context, and there is a proxy applied to that object to check if all registers have been registered, then it will emit the `circuitReady` event. 

However, the update of `quantumCircuitConfig` object does not mean that the corresponding initialisation script has been executed, thus logically it should be put after the shell execute command to make sure that the script has been executed, then wait for quantum circuit to be ready.

Also if I changed the code to
```
await shell.execute(script, async (err) => {
  if (err) node.error(err);
  else {
    await circuitReady();
    send(output);
  }
});
```
it will also block the process and I'm not sure about the reason. The current code is the only way I can make it work properly.

Reply:
Reverted to previous version